### PR TITLE
KTIJ-26406 False negative "Unused unary operator" with more than one operator on same line

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KTIJ-26406

スクリーンショット 2023-07-27 14 35 10

### CODE REVIEW

File: plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/inspections/UnusedUnaryOperatorInspection.kt [hunk 1]
```diff
@@ -39,31 +40,30 @@ class UnusedUnaryOperatorInspection : AbstractKotlinInspection() {
         val operationToken = prefix.operationToken
         if (operationToken != KtTokens.PLUS && operationToken != KtTokens.MINUS) return
 
-        // hack to fix KTIJ-196 (unstable `USED_AS_EXPRESSION` marker for KtAnnotationEntry)
-        if (prefix.isInAnnotationEntry) return
```
Comment (abelkov) [inline plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/inspections/UnusedUnaryOperatorInspection.kt:43]: This line is still needed, even though tests pass without it :( I will expand the comment.

**Review by abelkov — CHANGES_REQUESTED** [review]
Sorry, after your change there is a lot of new warnings on intellij-community project, please fix the inspection before we can commit this.

**Comment by abelkov** [comment]
Can you please add a test for the false positive case that was found in intellij-community?

**Comment by t-kameyama** [comment]
Added

**Review by abelkov — APPROVED** [review]

**Comment by vladimirdolzhenko** [comment]
Thanks a lot for the contribution

**Pull request closed without merging**